packages/webapp/pages/onboarding.tsx

```diff
@@ -0,0 +1,210 @@
+  [OnboardingFilteringTitle.Control]: 'Choose topics to follow',
+  [OnboardingFilteringTitle.V1]: 'What topic best describes you?',
+  [OnboardingFilteringTitle.V2]: 'Which topics resonate with you the most?',
+  [OnboardingFilteringTitle.V3]: `Pick the topics you'd love to dive into`,
+  [OnboardingFilteringTitle.V4]: 'Choose the topics you’re passionate about',
+};
+
+const Title = classed('h2', 'font-bold typo-title2');
+
+interface AuthProps {
+  isAuthenticating: boolean;
+  isLoginFlow: boolean;
+}
+
+const maxAuthWidth = 'tablet:max-w-[30rem]';
+
+const Container = classed(
+  'div',
+  'flex overflow-auto overflow-x-hidden absolute inset-0 items-center w-screen h-screen min-h-screen z-[100] bg-theme-bg-primary',
+);
+
+export function OnboardPage(): ReactElement {
+  const router = useRouter();
+  const { user, isAuthReady } = useAuthContext();
+  const [isFiltering, setIsFiltering] = useState(false);
+  const [finishedOnboarding, setFinishedOnboarding] = useState(false);
+  const [auth, setAuth] = useState<AuthProps>({
+    isAuthenticating: false,
+    isLoginFlow: false,
+  });
+  const { isAuthenticating, isLoginFlow } = auth;
+  const { onShouldUpdateFilters } = useOnboardingContext();
+  const {
+    onboardingFilteringTitle,
+    onboardingMinimumTopics,
+    onboardingV2,
+    isFeaturesLoaded,
+  } = useFeaturesContext();
+  const { trackEvent } = useAnalyticsContext();
+
+  const onClickNext = () => {
+    const screen = isFiltering ? OnboardingStep.Topics : OnboardingStep.Intro;
+
+    trackEvent({
+      event_name: AnalyticsEvent.ClickOnboardingNext,
+      extra: JSON.stringify({ screen_value: screen }),
+    });
+
+    if (!isFiltering) return setIsFiltering(true);
+
+    return setAuth({ isAuthenticating: true, isLoginFlow: false });
+  };
+
+  const formRef = useRef<HTMLFormElement>();
+  const title = versionToTitle[onboardingFilteringTitle];
+  const percentage = isAuthenticating ? 100 : 50;
+  const content = isAuthenticating
+    ? 'Once you sign up, your personal feed\nwill be ready to explore.'
+    : `Pick a few subjects that interest you.\nYou can always change these later.`;
+
+  const onSuccessfulTransaction = () => {
+    onShouldUpdateFilters(true);
+    setFinishedOnboarding(true);
+    router.push('/');
+  };
+
+  const isPageReady = isFeaturesLoaded && isAuthReady;
+
+  useEffect(() => {
+    if (!isPageReady) return;
+
+    if (user || onboardingV2 === OnboardingV2.Control) {
+      router.push('/');
+      return;
+    }
+
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_type: TargetType.MyFeedModal,
+      target_id: onboardingV2,
+      extra: JSON.stringify({
+        origin: OnboardingMode.Wall,
+        steps: [OnboardingStep.Topics],
+        mandating_categories: onboardingMinimumTopics,
+      }),
+    });
+  }, [
+    trackEvent,
+    isPageReady,
+    onboardingV2,
+    onboardingMinimumTopics,
+    router,
+    user,
+  ]);
```

Comment:
This should be split in two `useEffect`. One will call `trackEvent` and the second one will be in charge of redirect. That way you will have less dependencies to work with in both effects and they will also be easier to understand.

Reply:
Added this idea to web team meeting, think we would benefit from a useEffect hook that runs once for tracking purposes.

Reply:
We wouldn't want to have users with the same condition of redirecting to get the impression event. They are still going to have the same dependencies. I used to separate `useEffects` in the past too, but lesser `useEffects` are always better.

Reply:
But this will fire multiple times, then it's better to have a empty dependency array for now and exclude it.
Or add a state to ensure it only tracks once.

Reply:
Right. Yeah, let's add a state to ensure sending a single event. And create a hook in the future for these cases.